src/ast/mod.rs

```diff
@@ -3181,6 +3181,14 @@ pub enum Statement {
         representation: UserDefinedTypeRepresentation,
     },
     /// ```sql
+    /// CREATE TYPE <name> AS ENUM
+    /// ```
+    /// Note: this is a PostgreSQL-specific statement. See <https://www.postgresql.org/docs/current/sql-createtype.html>
+    CreateTypeAsEnum {
+        name: ObjectName,
+        labels: Vec<Ident>,
+    },
```

Comment:
I'm thinking it could make sense to include it as part of the existing `CreateType`, especially since its already an enum there wouldn't be as much of disadvantage in terms of backwards compatibility? then going forward we'd be able to extend it in a similar manner with future extensions to the statement. something like:
```rust
pub enum UserDefinedTypeRepresentation {
    Composite {
        attributes: Vec<UserDefinedTypeCompositeAttributeDef>,
    },
    // new variant
    Enum(CreateTypeEnum)
}
```

Reply:
Oh! I somehow missed that `UserDefinedTypeRepresentation` was an enum. I thought it was a struct which would be more invasive to change, hence the new `CreateTypeAsEnum`. But yeah, as an enum that's a much better place for it.

I ended up with:
```rust
pub enum UserDefinedTypeRepresentation {
    Composite {
        attributes: Vec<UserDefinedTypeCompositeAttributeDef>,
    },
    /// Note: this is PostgreSQL-specific. See <https://www.postgresql.org/docs/current/sql-createtype.html>
    Enum { labels: Vec<Ident> },
}
```
…so that `Enum` matches `Composite`. If you think it should be a real struct instead of the anonymous enum struct I can change it.